### ISSUE TITLE
Improve regex for versions in yaml

### ISSUE THREAD

**Issue by achronis** [sdmx-twg/sdmx-rest#161]
Hello

It seems the [regular expression](https://github.com/sdmx-twg/sdmx-rest/blob/master/api/sdmx-rest.yaml#L438) for `{versions}`
to allow some combinations like:

2~.*.-1
2~.*.+
2.-5.5
2..2*

Tested via nodejs.

```js
let reg=/^(\*\+|~|\*|(0|[1-9]\d*[\+\*~]?|[\+\*~]?)\.(0|[1-9]\d*[\+\*~]?|[\+\*~]?)\.?(0|[1-9]\d*[\+\*~]?|[\+\*~]?)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$/;

// all the following return true
reg.test('2~.*.-1'); 
reg.test('2.-5.5');
reg.test('2~.*.0');
 reg.test('2~.*.-1');
reg.test('2..2*');
```

**Comment by dosse** [comment]
The proposed regex syntax for versions in rest requests is:

`^(\+|~(\.0){0,2}|\*(\.0){0,2}|(0|[1-9]\d*)(\.(0|[1-9]\d*))?|(0|[1-9]\d*)(\.(0|[1-9]\d*)){2}(-(([A-Za-z-][A-Za-z0-9-]*)|([A-Za-z0-9-]+[A-Za-z-][A-Za-z0-9-]*)|(0|[1-9][0-9]*))(\.(([A-Za-z-][A-Za-z0-9-]*)|([A-Za-z0-9-]+[A-Za-z-][A-Za-z0-9-]*)|(0|[1-9][0-9]*)))*)?|\+\.0\.0|[1-9]\d*(\.\+\.0|\+(\.(0|[1-9]\d*)){2}|\.(0|[1-9]\d*)(\.\+|\+\.(0|[1-9]\d*)|\.(0|[1-9]\d*)\+))|(0|[1-9]\d*)((\*|~)(\.(0|[1-9]\d*)){1,2}|\.(\*|~)(\.0)?|\.(0|[1-9]\d*)((\*|~)(\.(0|[1-9]\d*))?|\.(0|[1-9]\d*)?(\*|~)))(,(\+|~(\.0){0,2}|\*(\.0){0,2}|(0|[1-9]\d*)(\.(0|[1-9]\d*))?|(0|[1-9]\d*)(\.(0|[1-9]\d*)){2}(-(([A-Za-z-][A-Za-z0-9-]*)|([A-Za-z0-9-]+[A-Za-z-][A-Za-z0-9-]*)|(0|[1-9][0-9]*))(\.(([A-Za-z-][A-Za-z0-9-]*)|([A-Za-z0-9-]+[A-Za-z-][A-Za-z0-9-]*)|(0|[1-9][0-9]*)))*)?|\+\.0\.0|[1-9]\d*(\.\+\.0|\+(\.(0|[1-9]\d*)){2}|\.(0|[1-9]\d*)(\.\+|\+\.(0|[1-9]\d*)|\.(0|[1-9]\d*)\+))|(0|[1-9]\d*)((\*|~)(\.(0|[1-9]\d*)){1,2}|\.(\*|~)(\.0)?|\.(0|[1-9]\d*)((\*|~)(\.(0|[1-9]\d*))?|\.(0|[1-9]\d*)?(\*|~)))))*)$`

Unfortunately, I couldn't make it shorter.   
Based on the definitions listed in https://github.com/sdmx-twg/sdmx-rest/issues/114, the proposed syntax allows for the following combinations ("[0]" stands for any non-negative number, "[1]" stands for any positive number, "-XXXX" is the semantic extension):

```text
[0]
[0].[0]
[0].[0].[0]
[0].[0].[0]-XXXXX
+
+.0.0
[1]+.[0].[0]
[1].+.0
[1].[0]+.[0]
[1].[0].+
[1].[0].[0]+
~
~.0
~.0.0
[0]~.[0]
[0]~.[0].[0]
[0].~
[0].~.0
[0].[0]~
[0].[0]~.[0]
[0].[0].~
[0].[0].[0]~
*
*.0
*.0.0
[0]*.[0]
[0]*.[0].[0]
[0].*
[0].*.0
[0].[0]*
[0].[0]*.[0]
[0].[0].*
[0].[0].[0]*
any of the above combinations separated by comma
```

Whenever only two version parts are provided, then only 2-part versions are matched.
Whenever at least 3 version parts are provided, then only 3-part versions are matched.
Note that the following version syntax examples would be wrong and are not allowed:
```
+.2.3
1.+.3
~.2
~.2.3
1.~.3
*.2
*.2.3
1.*.3
```
This is because version parts after a part that is +, ~ or * cannot be any positive number as it doesn't make sense.   
`+.0` is not allowed because + stands for stable versions that must necessarily have 3 version parts.

I will prepare a merge request for the schema update and also include more details in the documentation.

**Comment by sosna** [comment]
Fixed in v2.1.0
